src/utils/session.ts

```diff
@@ -42,8 +40,24 @@ export function joinRoom() {
     return;
   }
   peer.value?.destroy();
-  peer.value = new Peer(PEER_JS_OPTIONS);
-  console.info(`Join room ${roomName} as ${playerName}.`);
+  peer.value = new Peer(hostId.value, PEER_JS_OPTIONS);
+  const errorHandler = (e: PeerError<string>) => {
+    switch (e.type) {
+      case "unavailable-id":
```

Comment:
Does this mean if the roomName does not exist, we will create it and join anyways?

I am not sure is it a good idea to handle business logic in a error handler.

Reply:
Actually, it does not provide any explicit API for checking if an ID is taken and we can only know that using error event in an asynchronized way (this error is raised after it checks with the signal server).